src/main/webapp/view/transcript/transcript.js

```diff
@@ -113,60 +70,101 @@ function seekTranscript(currentTime) {
+
+    const template = document.getElementById(TRANSCRIPT_TEMPLATE);
+    this.attachShadow({mode: 'open'});
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.updateTemplateSlot(TRANSCRIPT_SLOT_TIME_RANGE, timestampRange);
+    this.updateTemplateSlot(TRANSCRIPT_SLOT_CONTENT, transcriptLine.content);
+
+    this.transcriptLine = transcriptLine;
   }
-  transcriptLineLiElement.classList.add(BOLD_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(DEFAULT_FONT_WEIGHT);
-}
 
-/**
- * Removes bold from the text in `transcriptLineLiElement` if it
- * is currently bolded.
- */
-function removeBold(transcriptLineLiElement) {
-  if (!isBolded(transcriptLineLiElement)) {
-    return;
+  /**
+   * Updates the template slot `slotName` with `slotValue`.
+   */
+  updateTemplateSlot(slotName, slotValue) {
+    const span = document.createElement('span');
+    span.innerText = slotValue;
+    span.slot = slotName;
+    this.appendChild(span);
   }
-  transcriptLineLiElement.classList.add(DEFAULT_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(BOLD_FONT_WEIGHT);
-}
 
-/** Returns true if`transcriptLineLiElement` is bolded. */
-function isBolded(transcriptLineLiElement) {
-  return transcriptLineLiElement.classList.contains(BOLD_FONT_WEIGHT);
-}
+  /** Returns true if the element is already bolded. */
+  isBolded() {
+    return this.classList.contains(BOLD_FONT_WEIGHT);
+  }
 
-/**
- * Returns true if `currentTimeMs` is within the time range for
- * the current transcript line.
- */
-function isWithinCurrentTimeRange(currentTimeMs) {
-  return currentTranscriptLine.startTimestampMs <= currentTimeMs &&
-      currentTimeMs <= currentTranscriptLine.endTimestampMs;
-}
+  /**
+   * Bolds the element if it is not already
+   * bolded.
+   */
+  addBold() {
+    if (this.isBolded()) {
+      return;
+    }
+    this.classList.add(BOLD_FONT_WEIGHT);
+    this.classList.remove(DEFAULT_FONT_WEIGHT);
+  }
 
-/**
- * Scrolls `transcriptLine` to the top of the transcript area.
- * */
-function scrollToTopOfTranscript(transcriptLine) {
-  const transcriptContainer = document.getElementById(TRANSCRIPT_CONTAINER);
-  const ulElementOffset = transcriptLine.parentElement.offsetTop;
-  transcriptContainer.scrollTop = transcriptLine.offsetTop - ulElementOffset;
+  /**
+   * Removes bold from the element if it
+   * is currently bolded.
+   */
+  removeBold() {
+    if (!this.isBolded()) {
+      return;
+    }
+    this.classList.add(DEFAULT_FONT_WEIGHT);
+    this.classList.remove(BOLD_FONT_WEIGHT);
+  }
+
+  /**
+   * Returns true if `timeMs` is within the time range for
+   * this transcript line.
+   */
+  isWithinTimeRange(timeMs) {
+    return this.transcriptLine.startTimestampMs <= timeMs &&
+    timeMs <= this.transcriptLine.endTimestampMs;
+  }
+
+  /**
+   * Scrolls this transcript line to the top of the transcript area.
+   */
+  scrollToTopOfTranscript() {
```

Comment:
if this class is about a TranscriptLineElement, then how can it scroll to the top of the transcript? That would implement multiple transcript lines.